### ISSUE TITLE
tests fail when git python module isn't installed

### ISSUE THREAD

**Issue by paper42** [PyCQA/pylint#5486]
### Bug description

Run pytest without having the git python module installed, they will fail on `import git`:

https://github.com/PyCQA/pylint/blob/5e9d20dc32768e873ab84bc4ed0b489fdda40672/pylint/testutils/primer.py#L1-L10

cc @Pierre-Sassoulas

### Command used

```shell
pytest
```

### Pylint output

```shell
-
```

### Expected behavior

Tests which require git python module are skipped on systems without this module.

### Pylint version

```shell
pylint 2.12.2
astroid 2.9.0
Python 3.10.0 (default, Oct 25 2021, 19:14:18) [GCC 10.2.1 20201203]
```

**Comment by Pierre-Sassoulas** [comment]
We [added git python to the tests requirements](https://github.com/PyCQA/pylint/blob/main/requirements_test_min.txt#L6) when we added primer, but they are indeed used in ``pylint.testutils``. So they should be a pylint runtime dependency.

I'd hate to do that though, most user do not need testutils and only use pylint as a linter a not as a lib that they extend.

I think we could create a separate package ``pylint-testutils`` so it does not pollute pylint's dependencies. Another advantage would be that we can do major when we break something in ``pylint.testutil`` instead of what we're doing right now (consider that it's not a public API). What do you think @DanielNoord @cdce8p ?

**Comment by cdce8p** [comment]
> So they should be a pylint runtime dependency.

`pylint.testutils` isn't a runtime dependency. It works just fine for Home-Assistant. Only if you want to use the pylint test interface is it required.

A separate package is a lot of work even if we automate most of it. I would suggest `extras_require` for it. Users could then do install it like this. (We can obviously change the name.)
```shell
pip install pylint[test-lib]
```
https://setuptools.pypa.io/en/latest/userguide/dependency_management.html#optional-dependencies

**Comment by Pierre-Sassoulas** [comment]
``pip install pylint[testutil]`` does sound like the best of both world, I like that.
